overlord/ifacestate/handlers.go

```diff
@@ -572,6 +572,19 @@ func (m *InterfaceManager) doAutoConnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
+	snapInfo, err := snap.ReadInfo(snapsup.Name(), snapsup.SideInfo)
```

Comment:
any reason not to do this inside WaitRestart itself ?

Reply:
Yeah, makes sense. Makes the testing slightly longer but it makes sense.